docs/reference/search/search-your-data/retrievers-examples.asciidoc

```diff
@@ -0,0 +1,423 @@
+[[retrievers-examples]]
+=== Retrievers examples
+
+To demonstrate the full functionality of retrievers, these examples require access to a <<semantic-reranking-models,semantic reranking model>> set up using the <<inference-apis,Elastic inference APIs>>.
+
+To begin with, we'll set up the necessary services and have them in place for later use.
+
+[source,js]
```

Comment:
As much as possible we should use [source, console] and [test snippets](https://github.com/elastic/elasticsearch/tree/main/docs#test-code-snippets). 

See this [example doc](https://github.com/elastic/elasticsearch/blob/main/docs/reference/search/retriever.asciidoc?plain=1).

You'll need to skip stuff that requires ML or semantic reranking, but everything else should be tested.

Reply:
This could be a follow up PR to not block but one of the big strengths of the ES docs is that they're tested :)

Reply:
Yeap, that's fair! Will follow up with another PR :)